### PR TITLE
bacpop-121 Fetch initial project data on load

### PR DESCRIPTION
**Pull request by EmmaLRussell**
This branch fetches available project data from back end and loads into front end state. Only the data currently provided by https://github.com/bacpop/beebop_py/pull/25 - targeted by the dependencies script in this PR - is currently available. 
To test - run up the app with `scripts/run_test` - create and run a project. Return to Home page and load the project. You should see in Network tab that sketch data is returned in the request, and the placeholder AMR and filenames should be visible in the Project page.  

Also started putting in a Project Loading view - this is currently only momentarily visible while the data is being fetched but it may take longer to load projects once we're loading everything. This displays a spinner and an array of messages describing load progress. This is a bit experimental, and with one eye on improving the load process in HINT, so might not be how we eventually manage user feedback during loading.

### CODE REVIEW

File: app/client/src/components/SavedProjects.vue [hunk 1]
```diff
@@ -8,27 +8,41 @@
...
-        this.getSavedProjects();
+const savedProjects = computed(() => store.state.savedProjects);
+function loadProject(project: SavedProject) {
+    store.dispatch("loadProject", project);
+    router.push("/project");
+}
+
+function loadProjectFromKey(project: SavedProject, keyCode: number) {
+    // load on Enter
+    if (keyCode === 13) {
```
Comment (richfitz) [inline app/client/src/components/SavedProjects.vue:39]: I can't work out from the interface where this is triggered?

**Review by richfitz — COMMENTED** [review]
I was surprised that this leaves you back in the "Start Analysis" page, and that running the analysis was possible and necessary - is that because the corresponding beebop_py change does not return enough back to make it clear that the analysis has been run?

Otherwise changes look fine and happy to keep moving on while I remember how it all fits together

**Review by richfitz — COMMENTED** [review]
I was surprised that this leaves you back in the "Start Analysis" page, and that running the analysis was possible and necessary - is that because the corresponding beebop_py change does not return enough back to make it clear that the analysis has been run?

Otherwise changes look fine and happy to keep moving on while I remember how it all fits together

File: app/client/screenshot.png
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply app/client/screenshot.png:1]: Nope! deleted now

File: app/client/src/components/LoadingSpinner.vue [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            <g transform="rotate(270)">
+                <circle cx="0" cy="0" r="7" fill="#42b983" fill-opacity="0.25" :transform="transforms[6]">
+                    <animateTransform attributeName="transform" type="scale" begin="-0.125s" values="1.1 1.1;1 1"
+                                      keyTimes="0;1" dur="1s" repeatCount="indefinite"></animateTransform>
+                    <animate attributeName="fill-opacity" keyTimes="0;1" dur="1s" repeatCount="indefinite" values="1;0"
+                             begin="-0.125s"></animate>
+                </circle>
+            </g>
+        </g>
+        <g transform="translate(71.21320343559643,28.78679656440357)">
```
[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply app/client/src/components/LoadingSpinner.vue:75]: 😱

**Comment by EmmaLRussell** [comment]
> I was surprised that this leaves you back in the "Start Analysis" page, and that running the analysis was possible and necessary - is that because the corresponding beebop_py change does not return enough back to make it clear that the analysis has been run?

It doesn't do yet, that will be in future tickets.

**Pull request merged**